File: python/강의/풀이/탐색 & 시뮬레이션/6. 격자판 최대합.py
import sys

sys.stdin = open("input.txt")

n = int(input())
nums = [list(map(int, input().split())) for _ in range(n)]
M = 0
for i in range(n):  # 가로 합
    sum1 = 0
    sum2 = 0
    for j in range(n):
        sum1 += nums[i][j]
        sum2 += nums[j][i]
    M = max(sum1, sum2, M)

for i in range(n):
    sum1 = 0
    sum2 = 0
    sum1 += nums[i][i]
    sum2 += nums[i][n-i-1]
    M = max(sum1, sum2, M)

print(M)

File: JS/인프런/myCode/1, 2차원 배열 탐색/6. 격자판 최대합.js
function solution(arr) {
    let answer = Number.MIN_SAFE_INTEGER;
    let n = arr.length;
    let sum1 = sum2 = 0;

    for (let i = 0; i < arr.length; i++) {
        sum1 = sum2 = 0;
        for (let j = 0; j < arr.length; j++) {
            sum1 += arr[i][j];
            sum2 += arr[j][j];
        }
        answer = Math.max(answer, sum1, sum2)
    }

    sum1 = sum2 = 0;

    for (let i = 0; i < arr.length; i++) {
        sum1 += arr[i][i]
        sum2 += arr[n - i - 1][i]
    }
    answer = Math.max(answer, sum1, sum2)
    return answer
}

let input = [[10, 13, 10, 12, 15],
[12, 39, 30, 23, 11],
[11, 25, 50, 53, 15],
[19, 27, 29, 37, 27],
[19, 13, 30, 13, 19]];

console.log(solution(input));
